### PR TITLE
Sets closed_at field on conversation update

### PR DESCRIPTION
**Pull request by jeepers3327**
### Description
* Add closed_at field to `conversations` table
* Update the changeset of `ChatApi.Conversations.Conversation` module to include `closed_at`
* Instead of setting the value of `closed_at` field after conversation update, I created a function inside `ChatApi.Conversations.Conversation` module to set `closed_at` depending on the `status` passed and include it in changeset pipeline.

### Issue

This handles #486 

## Checklist

- [X] Everything passes when running `mix test`
- [X] Ran `mix format`
- [X] No frontend compilation warnings

### CODE REVIEW

File: test/chat_api/conversations_test.exs [hunk 1]
```diff
@@ -286,6 +286,20 @@ defmodule ChatApi.ConversationsTest do
       refute conv2.archived_at
     end
 
+    test "update_conversation/2 sets the closed_at field based on updated status", %{
```
Comment (reichert621) [inline test/chat_api/conversations_test.exs:289]: hooray for tests 🎉 can we add one to `conversation_controller_text.exs` as well?

**Review by reichert621 — COMMENTED** [review]
looks great @jeepers3327! thanks for the quick turnaround on this 🚀 

just one small request -- would love to have a test in `conversation_controller_test.exs` that verifies this change works as expected too :)

File: test/chat_api/conversations_test.exs (hunk 1 above)
Comment (jeepers3327) [reply test/chat_api/conversations_test.exs:289]: Can I ask how should I perform the controller test for these changes? Do I need to add `closed_at` field in "basic.json" in `ChatApiWeb.ConversationView`? I'm not sure about this one.

Hope for your guidance on this one so I could perform the remaining tests.
Thank you.

Comment (reichert621) [reply test/chat_api/conversations_test.exs:289]: yup you can add `closed_at` to both `basic.json` and `expanded.json`, and then make sure this value is changed correctly when the `status` field is updated

Comment (jeepers3327) [reply test/chat_api/conversations_test.exs:289]: Would it be ok to just assert the `closed_at` field like  `assert closed_at != nil` if status is "closed" while ` assert closed_at == nil` if status is "open" ?

Comment (reichert621) [reply test/chat_api/conversations_test.exs:289]: yup that's fine 👍

**Review by reichert621 — APPROVED** [review]
looks great, thank you @jeepers3327! 🎉

**Pull request merged**